### PR TITLE
Change exception::invalid_prop -> not_supported

### PR DESCRIPTION
**Pull request by zhoonit**
Change invalid_prop -> not supported for generality

See also #374 

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: nntrainer/include/layer.h [hunk 1]
```diff
@@ -263,7 +263,7 @@ class Layer {
    * but throws error when @a type is invalid
    * @exception std::out_of_range     when property type is not valid for the
    * particular layer
-   * @exception exception::invalid_property invalid argument
+   * @exception exception::not_supported invalid argument
```
Comment (kparichay) [inline nntrainer/include/layer.h:266]: update description as well?

File: nntrainer/include/nntrainer_error.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ namespace exception {
 /**
  * @brief derived class of invalid argument to represent property is invalid
```
Comment (kparichay) [inline nntrainer/include/nntrainer_error.h:47]: ditto.

File: nntrainer/include/layer.h (hunk 1 above)
Comment (zhoonit) [reply nntrainer/include/layer.h]: Updated :)

File: nntrainer/include/nntrainer_error.h (hunk 2 above)
Comment (zhoonit) [reply nntrainer/include/nntrainer_error.h]: ditto.

**Pull request merged**